v2/spanner-to-sourcedb/src/main/java/com/google/cloud/teleport/v2/templates/models/PreparedStatementValueObject.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright (C) 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.google.cloud.teleport.v2.templates.models;
+
```

Comment:
Could we consider using [AutoValue](https://chromium.googlesource.com/external/github.com/google/auto/+/auto-value-1.6.2/value/userguide/index.md) to build this class?

AutoValue work very well with dataflow - Especially by the fact that they generate the `equals` and `hash` function for you. They also memoize the hash and make comparisons quite fast (which is a plus for dataflow pipelines)

An example of a similar autovalue class could be referred to [here](https://github.com/GoogleCloudPlatform/DataflowTemplates/blob/d9243b5392d659df6ada2d947a43a088c56a15f1/v2/sourcedb-to-spanner/src/main/java/com/google/cloud/teleport/v2/source/reader/io/cassandra/rowmapper/CassandraFieldMapper.java)

Reply:
Sure @VardhanThigle

Reply:
Address the comment and used [AutoValue](https://chromium.googlesource.com/external/github.com/google/auto/+/auto-value-1.6.2/value/userguide/index.md) between there is no need of serializable so we can resolved this comment